include/zephyr/drivers/video-controls.h

```diff
@@ -107,18 +227,307 @@ enum video_power_line_frequency {
+ * Within these modes the camera determines best exposure, aperture, focusing, light metering,
+ * white balance and equivalent sensitivity. The controls of those parameters are influenced by
+ * the scene mode control. An exact behavior in each mode is subject to the camera specification.
+ * When the scene mode feature is not used, this control should be set to
+ * @ref VIDEO_SCENE_MODE_NONE to make sure the other possibly related controls are accessible.
+ */
+#define VIDEO_CID_SCENE_MODE (VIDEO_CID_CAMERA_CLASS_BASE + 26)
+enum video_scene_mode {
+	VIDEO_SCENE_MODE_NONE = 0,
+	VIDEO_SCENE_MODE_BACKLIGHT = 1,
+	VIDEO_SCENE_MODE_BEACH_SNOW = 2,
+	VIDEO_SCENE_MODE_CANDLE_LIGHT = 3,
+	VIDEO_SCENE_MODE_DAWN_DUSK = 4,
+	VIDEO_SCENE_MODE_FALL_COLORS = 5,
+	VIDEO_SCENE_MODE_FIREWORKS = 6,
+	VIDEO_SCENE_MODE_LANDSCAPE = 7,
+	VIDEO_SCENE_MODE_NIGHT = 8,
+	VIDEO_SCENE_MODE_PARTY_INDOOR = 9,
+	VIDEO_SCENE_MODE_PORTRAIT = 10,
+	VIDEO_SCENE_MODE_SPORTS = 11,
+	VIDEO_SCENE_MODE_SUNSET = 12,
+	VIDEO_SCENE_MODE_TEXT = 13,
+};
+
+/** This control locks or unlocks the automatic focus, exposure and white balance.
+ * The automatic adjustments can be paused independently by setting their lock bit to 1.
+ * The camera then retains the settings until the lock bit is cleared.
+ * When a given algorithm is not enabled, drivers return no error.
+ * An example might be an application setting bit @ref VIDEO_LOCK_WHITE_BALANCE when the
+ * @ref VIDEO_CID_AUTO_WHITE_BALANCE control is off (set to 0).
+ * The value of this control may be changed by exposure, white balance or focus controls.
+ */
+#define VIDEO_CID_3A_LOCK (VIDEO_CID_CAMERA_CLASS_BASE + 27)
+enum video_3a_lock {
+	VIDEO_LOCK_EXPOSURE = BIT(0),
+	VIDEO_LOCK_WHITE_BALANCE = BIT(1),
+	VIDEO_LOCK_FOCUS = BIT(2),
+};
+
+/** Starts single auto focus process.
+ * Setting this control when @ref VIDEO_CID_FOCUS_AUTO is on (set to 1) is undefined.
+ * Drivers should ignore such requests.
+ */
+#define VIDEO_CID_AUTO_FOCUS_START (VIDEO_CID_CAMERA_CLASS_BASE + 28)
+
+/** Aborts automatic focusing started with @ref VIDEO_CID_AUTO_FOCUS_START control.
+ * Setting this control when @ref VIDEO_CID_FOCUS_AUTO is on (set to 1) is undefined.
+ * Drivers should ignore such requests.
+ */
+#define VIDEO_CID_AUTO_FOCUS_STOP (VIDEO_CID_CAMERA_CLASS_BASE + 29)
+
+/** The automatic focus status.
+ * This is a read-only control.
+ * Setting @ref VIDEO_LOCK_FOCUS lock bit of the @ref VIDEO_CID_3A_LOCK control may stop updates
+ * of the @ref VIDEO_CID_AUTO_FOCUS_STATUS control value.
+ */
+#define VIDEO_CID_AUTO_FOCUS_STATUS (VIDEO_CID_CAMERA_CLASS_BASE + 30)
+enum video_auto_focus_status {
+	VIDEO_AUTO_FOCUS_STATUS_IDLE = 0,
+	VIDEO_AUTO_FOCUS_STATUS_BUSY = BIT(0),
+	VIDEO_AUTO_FOCUS_STATUS_REACHED = BIT(1),
+	VIDEO_AUTO_FOCUS_STATUS_FAILED = BIT(2),
+};
+
+/** Determines auto focus distance range for which lens may be adjusted. */
+#define VIDEO_CID_AUTO_FOCUS_RANGE (VIDEO_CID_CAMERA_CLASS_BASE + 31)
```

Comment:
REMOVE? Used in only one Linux driver:
`drivers/media/i2c/s5c73m3/s5c73m3-ctrls.c`